src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -135,6 +135,38 @@ private void OnDispatchedToUI(object sender, UIDispatcherEventArgs e)
         /// </summary>
         public event Action<PortModel> PortDisconnected;
 
+        public class NodeExecutionEventArgs : EventArgs
+        {
+            public Guid GUID { get; private set; }
+            public object Data { get; private set; }
+
+            public NodeExecutionEventArgs(NodeModel model, object data)
+            {
+                this.GUID = model.GUID;
+                this.Data = data;
+            }
+        }
+
+        /// <summary>
+        /// Event triggered before a node is executed.
+        /// </summary>
+        public event Action<NodeModel, NodeExecutionEventArgs> NodeExecutionBegin;
+
+        internal void OnNodeExecutionBegin(object data)
+        {
+            NodeExecutionBegin?.Invoke(this, new NodeExecutionEventArgs(this, data));
```

Comment:
it appears this passes the nodes evaluated output to this handler? 
Can you confirm that in one of the tests?